### PR TITLE
Fix #9067: Redirect signed-out users to resource URL without lessonId or classId

### LINKED ISSUE #9067: Signed-out user can access a resource URL containing class and lesson ids when users are allowed to explore resources without signing in
## Observed behavior

With this device settings

device

I can access a lesson resource page as a signed-out user when I know its URL via pasting it to the address bar, e.g. _/en/learn/#/topics/c/69b6ce4071fd54fd82fab705fc89797f?lessonId=ca87af39b5a025cd51d8b87ff2ce95db&classId=68fdf5eca4d9e18cc31b3fe9ef897e9f_ (note `lessonId` and `classId` query parameters that are used for resources in the class context)

page

## Expected behavior

Even though users are allowed to explore resources without signing in, giving access to lesson resources still doesn't make sense for anonymous users. For signed-out users, it should rather redirect them to the same resource but outside of the class context (e.g. _/en/learn/#/topics/c/69b6ce4071fd54fd82fab705fc89797f_). If there are some technical limitations on the routing level for doing this, we should at least check that we account for this URL when deciding between lesson and topic contexts of a resource as @rtibbles noted:

> I think the issue I can imagine is that we use the lesson id to contextualise the "more in" side panel and the completion modal too. So at the very least we should be adding extra checks for that as to whether the user is logged in.

## User-facing consequences

I haven't noticed any user-facing problems related to this except that the URL might be confusing a bit and as described above, it has the potential for causing problems with lesson/topic contexts.

## Steps to reproduce

1. Select "Allow users to explore resources without signing in" in device settings as an admin
2. Obtain a URL of a resource from a lesson as a learner
3. Sign out
4. Paste the URL to the browser address bar

## Context

- Kolibri version: Kolibri 0.15.1.dev0+git.20220127215827
- Operating system: Ubuntu 20.04.3 LTS
- Browser: Version 97.0.4692.99 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by GSAprod**
## Summary

When the "Allow users to explore resources without signing in" setting is enabled, if a signed-out user accesses a resource page with the `lessonId` or `classId` query parameters, it auto-redirects to the same page of the topic that does not contain these parameters.

Therefore, an check was added onto the `initializeState` method of the `TopicsContentPage.vue` file to verify if the user is not logged in. If so, it also checks for the existence of the parameters mentioned before in the URL. If at least one of these parameters exist, the user is redirected, using the `redirectBrowser` function, to the same page, but with the identifiers removed.

This change prevents problems regarding lesson and topic contexts from happening.

## References

Fixes #9067 

## Reviewer guidance

In order to reproduce this issue, login as a student and navigate to a resource from one lesson. The URL of the content should contain the `classId` and `lessonId` keywords in its parameters. Copy the URL, then log out.

When signed out, if the URL is pasted onto the search bar, the app should redirect to an URL that does not contain these keywords.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by nucleogenesis — COMMENTED** [review]
Code looks good to me and a quick test of it seems to work as intended. I'll leave final approval to @pcenov after his QA

**Comment by pcenov** [comment]
Thanks @GSAprod I confirm that this is implemented exactly as specified above!
@MisRob  while regression testing I noticed that this fixes only the case where the user is signed-out. If I am signed in as a different learner or an admin and I copy-paste the URL with the `lessonId` and `classId` keywords then the resource is not loaded correctly. So to have a comprehensive fix of the issue described in https://github.com/learningequality/kolibri/issues/9067 we should handle this scenario as well.

Here's a video of what I am observing:

https://github.com/learningequality/kolibri/assets/79847249/0bd8813f-a75b-40ad-84c6-b1098a1a101e

**Comment by MisRob** [comment]
> @MisRob while regression testing I noticed that this fixes only the case where the user is signed-out. If I am signed in as a different learner or an admin and I copy-paste the URL with the lessonId and classId keywords then the resource is not loaded correctly. So to have a comprehensive fix of the issue described in https://github.com/learningequality/kolibri/issues/9067 we should handle this scenario as well.

Thank you @pcenov for thorough testing, yes we'd ideally fix that as well for sure. In regards to this particular PR, I'd leave it up to @GSAprod - please let us know if you'd like to add some more fixes or if it'd be best to have a follow-up issue that someone else can pick up. I don't think it blocks merging this PR since it's definitely an improvement that fixes what was specified in the original issue.

**Comment by GSAprod** [comment]
Hi there!
I've explored potential solutions for the current issue. As I understand it, resolving this problem may involve querying the server API to verify if the logged-in user is enrolled in a class and if the lesson is accessible to the user. If one of these conditions isn't met, then the `redirectBrowser` function needs to be called the same way as when the user is not logged in.

Checking whether the user is an admin would be more simple, in this case, a new condition could be added, to check the value of `this.isAdmin` (though I believe the checks for class enrollment would also cover this case).

I'm not familiarized with the overall structure of the project, so learning how to implement this solution would take a while, and I'm currently not in a position to work on this issue, possibly only at a future time. Therefore, I would propose opening up a new issue with this specific problem, and marking this one as resolved.
Thanks!

**Review by pcenov — APPROVED** [review]
Thanks @GSAprod - I have opened a follow-up issue.

**Pull request merged**
